pulp_python/tests/functional/api/test_download_content.py

```diff
@@ -87,3 +87,30 @@ def test_full_pulp_to_pulp_sync(
     repo3 = python_repo_with_sync(remote3)
     summary2 = python_content_summary(repository_version=repo3.latest_version_href)
     assert summary2.present["python.python"]["count"] == PYTHON_MD_PACKAGE_COUNT
+
+
+@pytest.mark.parallel
+def test_pulp2pulp_sync_with_oddities(
+    python_repo_with_sync,
+    python_remote_factory,
+    python_publication_factory,
+    python_distribution_factory,
+    python_content_summary,
+):
+    """Test that Pulp can handle syncing packages with wierd names."""
+    remote = python_remote_factory(includes=["oslo.utils"], url="https://pypi.org")
+    repo = python_repo_with_sync(remote)
+    distro = python_distribution_factory(repository=repo)
+    # Test pulp 2 pulp full sync w/ live pypi apis
+    remote2 = python_remote_factory(includes=[], url=distro.base_url)
+    repo2 = python_repo_with_sync(remote2)
+    summary = python_content_summary(repository_version=repo2.latest_version_href)
+    assert summary.present["python.python"]["count"] > 0
+    # Test w/ publication
+    pub = python_publication_factory(repository=repo)
+    distro2 = python_distribution_factory(publication=pub)
+    remote3 = python_remote_factory(includes=[], url=distro2.base_url)
+    repo3 = python_repo_with_sync(remote3)
+    summary2 = python_content_summary(repository_version=repo3.latest_version_href)
```

Comment:
pulp_python can distribute with and without publications?
What is the difference? Can we deprecate and remove publications?

Reply:
Yep, ever since I added the live api to support twine uploads publications were no longer necessary. Publications are still around to remain backwards compatible. The main difference is that publications are slightly more performant since we don't have to dynamically generate the simple files during pip installs. During pulp2pulp syncs it really is no different then the live api. 

We probably could deprecate them, I would want to ask the community first if they would be okay with this. (Probably would also want to do some testing to see how much of a performance hit it would be).